### PR TITLE
bsdlib: Update BSD library for nrf9160

### PR DESCRIPTION
**Pull request by gbakke**
Library version 2ab769c1e67.

Signed-off-by: Glenn Ruben Bakke <glenn.ruben.bakke@nordicsemi.no>

### CODE REVIEW

File: bsdlib/include/bsd.h [hunk 1]
```diff
@@ -35,9 +59,9 @@ extern "C" {
  *          the system. The application shall not use any of the resources identified in
  *          bsd_platform.h.
  *
- * @note This API is designed to provide flexibility of being called from startup scripts.
+ * returns Zero on success or an error code otherwise.
```
Comment (metc) [inline bsdlib/include/bsd.h]: Should be `@return`

File: bsdlib/CHANGELOG.rst [hunk 2]
```diff
@@ -5,6 +5,21 @@ Changelog
...
 
+bsdlib 0.3.2
+************
+
+Updated library with various changes:
+
+* Changed socket option NRF_SO_RCVTIMEO to use nrf_timeval struct instead of uint32_t.
+* Improved PDN socket close (NRF_PROTO_PDN) function.
+* Added new errno values NRF_ENOEXEC, NRF_ENOSPC and NRF_ENETRESET.
+* Added return value on bsd_init() to indicate MODEM_DFU result codes or initialiation result.
```
Comment (metc) [inline bsdlib/CHANGELOG.rst]: Typo: initialization

**Comment by rlubos** [comment]
@b-gent Please review the docs at convenience.

File: bsdlib/CHANGELOG.rst [hunk 3]
```diff
@@ -5,6 +5,21 @@ Changelog
...
+* Corrected GNSS struct nrf_gnss_datetime_t to use correct size on ms member.
```
Comment (b-gent) [inline bsdlib/CHANGELOG.rst]:
```suggestion
* Corrected GNSS struct :c:type:`nrf_gnss_datetime_t` to use correct size on the ms member.
```

File: bsdlib/CHANGELOG.rst [hunk 4]
```diff
@@ -5,6 +5,21 @@ Changelog
...
+* Updated Modem DFU interface.
```
Comment (b-gent) [inline bsdlib/CHANGELOG.rst]:
```suggestion
* Updated modem DFU interface.
```

File: bsdlib/CHANGELOG.rst [hunk 5]
```diff
@@ -5,6 +5,21 @@ Changelog
...
+* Improved error reporting on network or connection loss.
+* Corrected value of NRF_POLLNVAL.
```
Comment (b-gent) [inline bsdlib/CHANGELOG.rst]:
```suggestion
* Corrected the value of ``NRF_POLLNVAL``.
```

**Comment by gbakke** [comment]
Thanks @b-gent! I have merged and squashed all the proposed changes.

**Comment by b-gent** [comment]
thanks @gbakke

**Comment by rlubos** [comment]
It was confirmed in https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/925 that this version works fine.

**Pull request merged**